lib/BackbeatConsumer.js

```diff
@@ -180,7 +181,24 @@ class BackbeatConsumer extends EventEmitter {
     }
 
     isReady() {
-        return this._consumerReady && this._zookeeperReady;
+        let zk = null;
+        let cs = null;
+        if (this._backlogMetrics) {
+            zk = this._zookeeper &&
+                this._zookeeper.getState().code ===
+                    zookeeper.State.SYNC_CONNECTED.code;
+        } else {
+            zk = true;
+        }
+
+        if (this._bootstrap) {
```

Comment:
This variable is only set in test mode. Are you sure there's nothing to check on the consumer in the normal case?